vim-engine/src/main/kotlin/com/maddyhome/idea/vim/vimscript/model/commands/SourceCommand.kt

```diff
@@ -48,13 +48,23 @@ data class SourceCommand(val range: Range, val modifier: CommandModifier, val ar
     return ExecutionResult.Success
   }
 
-  private fun expandUser(path: String): String {
-    if (path.startsWith("~")) {
+  private fun String.vimExpanded(): String {
+    var expanded = this
+    if (startsWith("~")) {
       val home = System.getProperty("user.home")
       if (home != null) {
-        return home + path.substring(1)
+        expanded = home + substring(1)
       }
     }
-    return path
+
+    val envRe = Regex("\\$[A-Za-z0-9_]+")
```

Comment:
I feel like this approach skips some cases. For example, the env variable can have brackets `${PATH}`, or the dollar can be escaped like `\$PATH`.
On windows, the variables use a different syntax `%PATH%`.
Also, I'm not sure if the env variable can start with the number.

This expansion seems to be a non-trivial task. I'd try to search some existing ways in java/kotlin to expand it. Or, maybe there is an existing utility in the IJ platform (you can ask about it in `#ij-dev` slack channel).

If nothing will work, let's try to write our own expander, but keeping in mind the cases I described. Probably, it would make sense to read the specification on the env variables (if there is any), or ask AI what cases there are.